core/src/main/java/org/elasticsearch/search/aggregations/bucket/filters/FiltersAggregatorBuilder.java

```diff
@@ -57,6 +58,8 @@ public FiltersAggregatorBuilder(String name, KeyedFilter... filters) {
 
     private FiltersAggregatorBuilder(String name, List<KeyedFilter> filters) {
         super(name, InternalFilters.TYPE);
+        // internally we want to have a fixed order of filters, regardless of the order of the filters in the request
+        Collections.sort(filters);
```

Comment:
I was wrong, I didn't realize Arrays.asList() doesn't copy but writes through to the underlying array (thanks for your other comment, or as Stan says: "You Know, I Learned Something Today..."). In that case we need to copy here (or better in the other ctor. I will change that.